src/macros.jl

```diff
@@ -0,0 +1,156 @@
+it is not empty and ignores some common info statements. Additional patterns
+that should be ignored can be passed as a list of strings or regular
+expressions.
+"""
+macro trixi_test_nowarn(expr, additional_ignore_content = String[])
+    quote
+        let fname = tempname()
+            try
+                ret = open(fname, "w") do f
+                    redirect_stderr(f) do
+                        $(esc(expr))
+                    end
+                end
+                stderr_content = read(fname, String)
+                if !isempty(stderr_content)
+                    println("Content of `stderr`:\n", stderr_content)
+                end
+
+                # Patterns matching the following ones will be ignored. Additional patterns
+                # passed as arguments can also be regular expressions, so we just use the
+                # type `Any` for `ignore_content`.
+                ignore_content = Any["[ Info: You just called `trixi_include`. Julia may now compile the code, please be patient.\n"]
+                append!(ignore_content, $additional_ignore_content)
+                for pattern in ignore_content
+                    stderr_content = replace(stderr_content, pattern => "")
+                end
+
+                # We also ignore simple module redefinitions for convenience. Thus, we
+                # check whether every line of `stderr_content` is of the form of a
+                # module replacement warning.
+                @test occursin(r"^(WARNING: replacing module .+\.\n)*$", stderr_content)
+                ret
+            finally
+                rm(fname, force = true)
+            end
+        end
+    end
+end
+
+"""
+    @test_trixi_include(elixir; l2=nothing, linf=nothing, RealT=Float64,
+                                atol=500*eps(RealT), rtol=sqrt(eps(RealT)),
+                                parameters...)
+
+Test Trixi by calling `trixi_include(elixir; parameters...)`.
+By default, only the absence of error output is checked.
+If `l2` or `linf` are specified, in addition the resulting L2/Linf errors
+are compared approximately against these reference values, using `atol, rtol`
+as absolute/relative tolerance.
+"""
+macro test_trixi_include(elixir, args...)
+    # Note: The variables below are just Symbols, not actual errors/types
+    local l2 = get_kwarg(args, :l2, nothing)
+    local linf = get_kwarg(args, :linf, nothing)
+    local RealT = get_kwarg(args, :RealT, :Float64)
+    if RealT === :Float64
+        atol_default = 500 * eps(Float64)
+        rtol_default = sqrt(eps(Float64))
+    elseif RealT === :Float32
+        atol_default = 500 * eps(Float32)
+        rtol_default = sqrt(eps(Float32))
+    elseif RealT === :Float128
+        atol_default = 500 * eps(Float128)
+        rtol_default = sqrt(eps(Float128))
+    elseif RealT === :Double64
+        atol_default = 500 * eps(Double64)
+        rtol_default = sqrt(eps(Double64))
+    end
```

Comment:
```suggestion
    local RealT_symbol = get_kwarg(args, :RealT, :Float64)
    RealT = getfield(@__MODULE__, RealT_symbol)
    atol_default = 500 * eps(RealT)
    rtol_default = sqrt(eps(RealT))
```
We [have that in TrixiShallowWater.jl](https://github.com/trixi-framework/TrixiShallowWater.jl/blob/cbc0bcc4b768b7888923f52a7a0a5d298018ca49/test/test_trixi.jl#L21-L24), which I think is nice because it's shorter, less redundant, and more generic.

Reply:
Looks very nice indeed.